prov/util/src/util_mem_monitor.c

```diff
@@ -205,6 +220,21 @@ int ofi_monitors_add_cache(struct ofi_mem_monitor **monitors,
 
 out:
 	pthread_mutex_unlock(&mm_lock);
+
+	/* If the cache has been assigned zero monitors, return an error. */
+	if (ret == FI_SUCCESS) {
+		for (iface = FI_HMEM_SYSTEM; iface < OFI_HMEM_MAX; iface++) {
+			if (cache->monitors[iface])
+				break;
```

Comment:
Add a success_cnt value and increment it if any monitors are added.  That can be used in place of iterating back over the monitor list again.

Reply:
Will do.

Reply:
Will do.